### PR TITLE
[systemtest] Tests for changing DeploymentStrategy in resources

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- New tests

### Description

In #3957 the possibility of configuring DeploymentStrategy was introduced, this PR adding tests for the feature. 

### Checklist

- [x] Write tests
- [ ] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectST#testConfigureDeploymentStrategy,ConnectS2IST#testConfigureDeploymentStrategy,HttpBridgeST#testConfigureDeploymentStrategy,MirrorMakerST#testConfigureDeploymentStrategy,MirrorMaker2ST#testConfigureDeploymentStrategy

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java [hunk 1]
```diff
@@ -284,6 +286,51 @@ void testHostAliases() {
         assertThat(output, containsString(etcHostsData));
     }
 
+    @Test
+    void testConfigureDeploymentStrategy() {
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3).done();
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java:291]: Kafka is created in `beforeAll`, do we need this "replace" ?

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java:291]: Yes, my bad... thanks for this spot!

**Review by scholzj — COMMENTED** [review]
TBH, I'mm not sure this is worth the time needed to run this. Because what does this actually test over the UTs we have for this? In general I do not see any bugs, but I also do not see much added value.

**Comment by im-konge** [comment]
@scholzj after "offline" conversation with you and @Frawless we decided to add `RELEASE` tag for running these kind of tests not so often like `REGRESSION`. But this will be "implemented" in future PRs, if it can be like this.

So we can merge these tests (if you don't have any other comments) and then do the rest.

Do you agree @Frawless and @scholzj ?

**Review by scholzj — APPROVED** [review]

**Review by Frawless — APPROVED** [review]
Fine by me

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectST#testConfigureDeploymentStrategy,ConnectS2IST#testConfigureDeploymentStrategy,HttpBridgeST#testConfigureDeploymentStrategy,MirrorMakerST#testConfigureDeploymentStrategy,MirrorMaker2ST#testConfigureDeploymentStrategy

**Pull request merged**